### PR TITLE
Support deploying SR-IOV network operator

### PR DESCRIPTION
**Pull request by adrianchiris**
This commit adds Helm chart to sriov-network-operator.

- Add this chart as an optional dependency to the main
  network-operator chart
- Add the needed chart files/crds/templates
- update README

### CODE REVIEW

File: deployment/network-operator/charts/sriov-network-operator/crds/sriovnetwork.openshift.io_sriovnetworknodepolicies_crd.yaml [hunk 1]
```diff
@@ -0,0 +1,112 @@
+apiVersion: apiextensions.k8s.io/v1
```
Comment (moshe010) [inline deployment/network-operator/charts/sriov-network-operator/crds/sriovnetwork.openshift.io_sriovnetworknodepolicies_crd.yaml:1]: probably I don't understand helm (maybe you can show me a demo) but  why do we need all the crds from SR-IOV operator?

Comment (adrianchiris) [reply deployment/network-operator/charts/sriov-network-operator/crds/sriovnetwork.openshift.io_sriovnetworknodepolicies_crd.yaml:1]: It is the intention of this chart to deploy all elements needed to use sriov network operator in a k8s cluster, so it includes its crds.

lets have an offline talk on this

File: deployment/network-operator/charts/sriov-network-operator/values.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  # limits:
+  #   cpu: 100m
+  #   memory: 128Mi
+  # requests:
+  #   cpu: 100m
+  #   memory: 128Mi
+
+# Image URIs for sriov-network-operator components
+images:
+  operator: adriancdockerhub/sriov-network-operator:latest
```
Comment (adrianchiris) [inline deployment/network-operator/charts/sriov-network-operator/values.yaml:35]: We should have sriov network operator work on vanilla k8s and have a tagged release with a container for it.

File: deployment/network-operator/charts/sriov-network-operator/templates/operator.yaml [hunk 3]
```diff
@@ -0,0 +1,68 @@
+apiVersion: apps/v1
```
Comment (Mmduh-483) [inline deployment/network-operator/charts/sriov-network-operator/templates/operator.yaml:1]: It's missing the license

File: deployment/network-operator/charts/sriov-network-operator/Chart.yaml [hunk 4]
```diff
@@ -0,0 +1,24 @@
+apiVersion: v2
+name: sriov-network-operator
+description: |
+  SR-IOV network operator configures and manages SR-IOV networks in the kubernetes cluster
+
+# A chart can be either an 'application' or a 'library' chart.
```
Comment (Mmduh-483) [inline deployment/network-operator/charts/sriov-network-operator/Chart.yaml:6]: Non-related comments

File: deployment/network-operator/charts/sriov-network-operator/values.yaml [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+      effect: "NoSchedule"
+  nodeSelector:
+    node-role.kubernetes.io/master: ""
+  nameOverride: ""
+  fullnameOverride: ""
+  # tag, if defined will use the given image tag, else Chart.AppVersion will be used
+  # tag
+
+resources: {}
+  # We usually recommend not to specify default resources and to leave this as a conscious
```
Comment (Mmduh-483) [inline deployment/network-operator/charts/sriov-network-operator/values.yaml:18]: Non-related comments

File: deployment/network-operator/charts/sriov-network-operator/templates/operator.yaml [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+            - name: NETWORK_RESOURCES_INJECTOR_IMAGE
+              value: ""
+            - name: OPERATOR_NAME
+              value: sriov-network-operator
+            - name: SRIOV_NETWORK_CONFIG_DAEMON_IMAGE
+              value: {{ .Values.images.sriovConfigDaemon | quote }}
+            - name: SRIOV_NETWORK_WEBHOOK_IMAGE
+              value: ""
+            - name: RESOURCE_PREFIX
+              value: openshift.io
```
Comment (Mmduh-483) [inline deployment/network-operator/charts/sriov-network-operator/templates/operator.yaml:52]: Consider changing it to nvidia.com

File: deployment/network-operator/charts/sriov-network-operator/templates/operator.yaml [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+          imagePullPolicy: IfNotPresent
+          env:
+            - name: WATCH_NAMESPACE
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.namespace
+            - name: SRIOV_CNI_IMAGE
+              value: {{ .Values.images.sriovCni }}
+            - name: SRIOV_INFINIBAND_CNI_IMAGE
+              value: {{ .Values.images.ibSriovCni | quote }}
```
Comment (Mmduh-483) [inline deployment/network-operator/charts/sriov-network-operator/templates/operator.yaml:40]: Is ` | quote` needed?

Comment (adrianchiris) [reply deployment/network-operator/charts/sriov-network-operator/templates/operator.yaml:40]: probably not needed

File: deployment/network-operator/charts/sriov-network-operator/templates/operator.yaml (hunk 6 above)
Comment (adrianchiris) [reply deployment/network-operator/charts/sriov-network-operator/templates/operator.yaml:52]: +1

File: deployment/network-operator/charts/sriov-network-operator/values.yaml (hunk 5 above)
Comment (adrianchiris) [reply deployment/network-operator/charts/sriov-network-operator/values.yaml:18]: These comments are auto generated by helm when it creates a package template, from my POV these can be left as is for the future as to ensure that resources are not explicitly specified

File: deployment/network-operator/charts/sriov-network-operator/Chart.yaml (hunk 4 above)
Comment (adrianchiris) [reply deployment/network-operator/charts/sriov-network-operator/Chart.yaml:6]: These comments are auto generated by helm when it creates a package template, from my POV these can be left as is

File: deployment/network-operator/charts/sriov-network-operator/values.yaml [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+  # limits:
+  #   cpu: 100m
+  #   memory: 128Mi
+  # requests:
+  #   cpu: 100m
+  #   memory: 128Mi
+
+# Image URIs for sriov-network-operator components
+images:
+  operator: quay.io/openshift/origin-sriov-network-operator:4.8
```
Comment (adrianchiris) [inline deployment/network-operator/charts/sriov-network-operator/values.yaml:45]: We should have all sriov-network-operator related images from official project repos
e.g not from openshift

**Review by moshe010 — APPROVED** [review]

**Pull request merged**